tests/unit/session.test.ts

```diff
@@ -1,42 +1,92 @@
-import { createRemoteJWKSet, jwtVerify, errors } from 'jose';
-import { Session } from '../../src/services';
-import ValidationError, { ValidationErrorNames } from '../../src/errors/validationError';
-import { httpStatusCodes } from '../../src/errors';
-
-// Mock jose functions
-jest.mock('jose', () => {
-  return {
-    jwtVerify: jest.fn() as jest.MockedFunction<typeof jwtVerify>,
-    createRemoteJWKSet: jest.fn() as jest.MockedFunction<typeof createRemoteJWKSet>,
-    errors: {
-      JWTClaimValidationFailed: class extends Error {},
-      JWTExpired: class extends Error {},
-      JWTInvalid: class extends Error {},
-    },
-  };
+// @ts-ignore
+import express from 'express';
+import { generateKeyPair, exportJWK, SignJWT, jwtVerify, createRemoteJWKSet, KeyLike } from 'jose';
+// @ts-ignore
+import http from 'http';
+import { httpStatusCodes, ValidationError } from '../../src/errors';
+import { ValidationErrorNames } from '../../src/errors/validationError';
+import SessionService from '../../src/services/sessionService';
+
+const app = express();
+const PORT = 8081;
+
+let privateKey: KeyLike;
+let publicKeyJwk: any;
+
+async function initializeKeys() {
+  const { privateKey: key, publicKey } = await generateKeyPair('RS256');
+  privateKey = key;
+  publicKeyJwk = await exportJWK(publicKey);
+  publicKeyJwk.kid = 'kid123';
+}
+
+// @ts-ignore
+app.get('/jwks', (req, res) => {
+  res.json({ keys: [publicKeyJwk] });
 });
 
+async function startJWKSserver() {
+  await initializeKeys();
+  return new Promise<http.Server>((resolve) => {
+    const server = app.listen(PORT, () => {
+      console.log(`JWKS server is running at http://localhost:${PORT}`);
+      resolve(server);
+    });
+  });
+}
+
+async function generateJWT(issuer: string, expiresIn: number, payload: Record<string, any>): Promise<string> {
```

Comment:
should be able to pass a private key here because please see Go SDK there I have a check with an invalid private key as well

Reply:
sorry it was right below :)